src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -60,39 +60,97 @@ CHIP_ERROR OTAImageProcessorImpl::Abort()
 
 CHIP_ERROR OTAImageProcessorImpl::Apply()
 {
-    ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_done(true)));
+    int err = dfu_target_done(true);
+    if (err == 0)
+    {
+        // schedule update of all possible targets by caling this function with argument -1
+        err = dfu_target_schedule_update(-1);
+    }
 
 #ifdef CONFIG_CHIP_OTA_REQUESTOR_REBOOT_ON_APPLY
-    return SystemLayer().StartTimer(
-        System::Clock::Milliseconds32(CHIP_DEVICE_CONFIG_OTA_REQUESTOR_REBOOT_DELAY_MS),
-        [](System::Layer *, void * /* context */) {
-            PlatformMgr().HandleServerShuttingDown();
-            k_msleep(CHIP_DEVICE_CONFIG_SERVER_SHUTDOWN_ACTIONS_SLEEP_MS);
-            sys_reboot(SYS_REBOOT_WARM);
-        },
-        nullptr /* context */);
+    if (err == 0)
+    {
+        return SystemLayer().StartTimer(
+            System::Clock::Milliseconds32(CHIP_DEVICE_CONFIG_OTA_REQUESTOR_REBOOT_DELAY_MS),
+            [](System::Layer *, void * /* context */) {
+                PlatformMgr().HandleServerShuttingDown();
+                k_msleep(CHIP_DEVICE_CONFIG_SERVER_SHUTDOWN_ACTIONS_SLEEP_MS);
+                sys_reboot(SYS_REBOOT_WARM);
+            },
+            nullptr /* context */);
+    }
+    else
+    {
+        return System::MapErrorZephyr(err);
+    }
 #else
-    return CHIP_NO_ERROR;
+    return System::MapErrorZephyr(err);
 #endif
 }
 
+CHIP_ERROR OTAImageProcessorImpl::SwitchToNextImage(ByteSpan & aRemainingData)
```

Comment:
Why not `(const ByteSpan & aRemainingData )`? data() and size() methods don't seem to modify an object.

Reply:
You're right.